### PR TITLE
[Back-end] Cart Model Schema

### PR DESCRIPTION
**Pull request by markjoshuacarmen**
## Ticket/s
[Cart Model](https://trello.com/c/VAQgfRkD/36-cart-model)

## Changes

- add cart schema 

## Screenshot

- N/A

## How to test?

- N/A

### CODE REVIEW

File: apps/api/src/cart/schemas/cart.schema.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+@Schema()
+export class Cart {
+  @Prop()
+  orderDate: string;
+
+  @Prop()
+  orderStatus: string;
+
+  @Prop()
+  orderedItems: Items;
```
Comment (TreshAnn) [inline apps/api/src/cart/schemas/cart.schema.ts]:
```suggestion
orderedItems:  Items[]
```

**Review by dandalandone — APPROVED** [review]
LGTM!

File: apps/api/src/cart/schemas/cart.schema.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { HydratedDocument } from 'mongoose';
+
+export type CartDocument = HydratedDocument<Cart>;
+
+@Schema()
+export class Items {
+  @Prop()
+  product_id: string;
```
Comment (TreshAnn) [inline apps/api/src/cart/schemas/cart.schema.ts]:
```suggestion
  @Prop({ type: ProductSchema, ref: Product.name })
  product_id: Product;
```

File: apps/api/src/cart/schemas/cart.schema.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { HydratedDocument } from 'mongoose';
+
```
Comment (TreshAnn) [inline apps/api/src/cart/schemas/cart.schema.ts:4]: Import Product schema here

**Review by TreshAnn — CHANGES_REQUESTED** [review]

File: apps/api/src/cart/schemas/cart.schema.ts (hunk 3 above)
Comment (markjoshuacarmen) [reply apps/api/src/cart/schemas/cart.schema.ts:4]: done thanks

**Review by TreshAnn — APPROVED** [review]

File: apps/api/src/cart/schemas/cart.schema.ts (hunk 2 above)
Comment (dandalandone) [reply apps/api/src/cart/schemas/cart.schema.ts]: @markjoshuacarmen no underscore, productID

File: apps/api/src/cart/schemas/cart.schema.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { HydratedDocument } from 'mongoose';
+import { Product, ProductSchema } from '../../products/schemas/products.schema';
+
+export type CartDocument = HydratedDocument<Cart>;
+
+@Schema()
+export class Items {
+  @Prop({ type: ProductSchema, ref: Product.name })
+  product_id: string;
```
Comment (dandalandone) [inline apps/api/src/cart/schemas/cart.schema.ts]: Make this productId

**Review by dandalandone — DISMISSED** [review]

File: apps/api/src/cart/schemas/cart.schema.ts (hunk 4 above)
Comment (markjoshuacarmen) [reply apps/api/src/cart/schemas/cart.schema.ts]: done change

**Review by DanDalanDone69 — APPROVED** [review]
LGTM!

**Pull request merged**